elastic/kibana
# [ResponseOps][Actions] Getting error about secrets not being saved when import a SO (Connector Saved Object)

Resolves [#134086](https://github.com/elastic/kibana/issues/134086)

## Summary

This is a special case error that seems to occur when exporting/importing connectors where authentication is optional, such as email and webhook connectors. On saved object export we drop `secrets`, and then for the connectors where there was auth or auth is required the user is given a warning that they need to set the secret values. 
For the connectors where auth is optional and not set, the secrets value needs to be defined. After import, it never gets redefined because the user does not need to enter any secret values, and then will throw a validation error when the connector is executed.
To fix this issue, I added `secrets: {}` to the saved object export if secrets are required or optional instead of completely dropping the field

### Checklist



### To verify

- Create an email connector where Authentication is toggled off
- Export the saved object, verify that the connector has `secrets: {}`
- Import the saved object, verify that no errors or warnings are shown
- Go to the test tab and run the imported connector. Verify that you do not see the following error
`error validating action type connector: secrets must be defined`
